Task 361-365: offenseval_dravidian, parsed_pdfs, agnews datasets

mjuluru opened the pull request.

swarooprm commented:
@amirrezamirzaei, wondering if you can help review these tasks?

amirrezamirzaei commented:
in task 361 explanation ` Good questions are expected ...` is repeated twice also I think it's not needed here at all.
task 362 & 363 have the same explanation are they the same?
change the name of file 361 to 363 so they don't all have the same name.

mjuluru commented:
Hi @swarooprm @amirrezamirzaei ,

I have addressed the suggested modifications and resubmitted the tasks @ 8.30 A.M. today.
It seems that all the test cases haven't run yet and is stuck in "workflow awaiting approval" status. Can you please have a look into it?

Thanks,
Mamatha J.

amirrezamirzaei commented:
hi @mjuluru 
don't worry about that @swarooprm just needs to approve the workflow. 
here are some other minor comments:

- in task 361-363 please change `their rating of offensiveness` to `their offensiveness` or something better.also please capitalize the language name. I don't understand what the not tamil class exactly is can you explain please.
- also some of the instances in 361-363 contain english language in addition to the specified language, please add this to the task definition.
- in task 364-365 please don't use `Things to avoid` and  `Emphasis & Caution` format.
- please use Classification instead of Classification Generation as the task categories.
- the task definitions provided in the readme file can be a little more descriptive for example for task 361 instead of
 `Generating classification to ...`  you can use `Given a statement in (Given language) classify how offensive the statement is.`

amirrezamirzaei commented:
also, I think it's better to break each task in 361-363 into multiple tasks.
for example task 361 can be two tasks
task1 : finding if a given statement is offensive or not
task2: finding if a given offensive statement is targeted or not

mjuluru commented:
Hi @amirrezamirzaei ,

I made the suggested changes to the tasks. Please check :
1. Changed the definitions and other text modifications
2. Modified the tasks 361 - 363 from rating of offensiveness to binary classification [offensive/not offensive]
3. Added extra descriptions in readme file

To answer your question, not-Tamil classification indicates that the input text doesn't belong to Tamil. It may be belong to any other language. 

Thanks,
Mamatha J.

amirrezamirzaei commented:
thanks for the changes a few other comments
1 - use classification instead of Classification Generation in the readme file also.
2- there are two descriptions for task 361 in the readMe file 
3- can you change the description(readMe) for 361-363 to `Given a statement in (Language), classify whether its offensive or not `
4- in task 361-363 `you're given statements in native Tamil language and Tamil written in English` would read a lot better if it was `In this task, you're given statements in native Tamil language. the statement can be written with the Tamil alphabet or the English alphabet.` also is the sentence `Some of these mentions refer to the same person, place, or thing` really needed here?(you also use this exact sentence in the other task I'm not sure what additional information its providing) you can get some ideas on a good definition for these task by looking at this [task](https://github.com/allenai/natural-instructions-expansion/blob/master/tasks/task286_olid_offense_judgment.json).
5- for task 364 you can get some ideas from this [task](https://github.com/allenai/natural-instructions-expansion/blob/master/tasks/task288_gigaword_summarization.json). use summarization as this task category. you can also use the same definition.
6- for task 365 please specify the classification labels.
7 - also please fix the file names ( changing classification generation to classification ... )

mjuluru commented:
Hi @amirrezamirzaei ,

I have implemented the changes as suggested. 
Please have a look into it.

Thanks,
Mamatha J.

amirrezamirzaei commented on tasks/README.md:
```diff
@@ -6,4 +6,9 @@ This directory contains the tasks that are part of this benchmark.
 `task1001_squad1.1_question_generation` | Generating guestions (based on SQuAD 1.1) | Question Generation  
-`task1002_squad1.1_answer_generation` | Generating answers to SQuAD 1.1 questions | Answer Generation
\ No newline at end of file
+`task1002_squad1.1_answer_generation` | Generating answers to SQuAD 1.1 questions | Answer Generation`Task361_offenseval_dravidian_classification_generation` | Generating classification to offensive dravidan questions | Classification Generation
```
please remove the first definition for 361

amirrezamirzaei commented on tasks/README.md:
```diff
@@ -6,4 +6,9 @@ This directory contains the tasks that are part of this benchmark.
+`Task361_tamil_offenseval_dravidian_classification` | Given a statement in Tamil, classify whether it is offensive or not | Classification
+`Task362_malayalam_offenseval_dravidian_classification` | Given a statement in Malayalam, classify whether it is offensive or not | Classification
+`Task363_kannada_offenseval_dravidian_classification` | Given a statement in Kannada, classify whether it is offensive or not | Classification
+`Task364_parsed_pdfs_summarization` | Given a text, generate title for it | Summarization
```
`generate a title for it`

amirrezamirzaei commented:
- in tasks 361-363 `Your job is to evaluate if the statement is offensive or not` has been repeated twice. also, you have used `\”` instead of `\"`. otherwise, these tasks look good to me.
- in task 364 `In this task, you are given a text of an article would read better if it was `In this task, you are given a part of an article
-in task 365 please change `given a text` to `given a short article`
- please provide the link of the paper or dataset in the source field for each task

mjuluru commented:
Hi @amirrezamirzaei ,

I have modified the files as suggested. Please have a look into it.

Thanks,
Mamatha J.

amirrezamirzaei commented:
thank you 
- task 361-363 looks good to me. just one last minor fix, I noticed that we use differents labels for language in tasks. please remove the `Input Language` and `Output Language` from the file and add "Input_language" , "Output_language", "instruction_language" instead. see this [task](https://raw.githubusercontent.com/allenai/natural-instructions-expansion/master/tasks/task272_europarl_translation.json) for more info. please use the same order for these field as in the task file. 
- task 364 looks good 
- task 365 please change `classify it based on its category...` to `classify the article based on its category...` (also change this in the readMe file please)

mjuluru commented:
Hi @amirrezamirzaei ,

I have performed the changes as suggested. Please have a look into it.

Thanks,
Mamatha J.

amirrezamirzaei commented:
thank you
- in tasks 361-363 please remove `This is a good example because the statement is offensive and is classified as offensive` from all of the positive examples (or similar sentences at the end of explanations)
- task 364-365 looks good.
- please address the comment on the readMe file. (the explanation of the last task)

mjuluru commented:
Hi @amirrezamirzaei ,

I addressed the changes as suggested.
Please review.

Thanks,
Mamatha J.

amirrezamirzaei commented:
thank you one final comment
- in the second negative example in task 362 shouldn't the output be `Offensive`?

amirrezamirzaei commented:
@swarooprm Looks good to me.

swarooprm commented:
Your HW2 is complete!
Would you like to submit the same tasks to our big project repo https://github.com/allenai/natural-instructions-expansion/pulls? You just need to fork that repo, rename tasks starting from the latest number in the repo and create PR there. This way, you can be part of our research on "learning from instructions instead of learning from examples (conventional machine learning)". This is purely optional and your HW2 grade does not depend on this @mjuluru ?

swarooprm commented:
@mjuluru, Just a reminder that, if you plan to submit these tasks to the main Github repo and join our research project, Monday (27th Sept) is the deadline. Again, this is optional and independent of your HW2 grade.
